### ISSUE TITLE
Variable pitch mode on just text after a header line

### ISSUE THREAD

**Issue by rolojf** [protesilaos/modus-themes#52]
Hola Prot

I want to share with you what I am doing. Maybe you see it interesting. Maybe suggest a better way to do this. Or maybe, you fix this and incorporate on your code. Or maybe not interesting.  Anyway, I think it is a good idea to share this with you.

I am not good at **elisp**. I don't know how to change this code _the right way_. This is why I didn't make a pull request to suggest a way to add this and save you some time.

I like using variable pitch mode on just the text after a header. Every other font (like header lines, tables, blocks, etc) Should be fixed pitch in order to look good and have aligned the tags and the indentation, the calendar, etc.

I have been using the following hack to make it look this way.

The relevant parts on my config files:
``` elisp
(set-face-attribute 'default nil :family "Iosevka Fixed" :height 140)
(set-face-attribute 'variable-pitch nil :family "Fira Sans Book" :height 0.96)
(set-face-attribute 'fixed-pitch nil :family (face-attribute 'default :family))

(add-hook 'text-mode-hook 'variable-pitch-mode)

(setq 
      modus-themes-mixed-fonts t
      modus-themes-headings
      '((1 . (overline variable-pitch bold 1.2))
        (2 . (rainbow variable-pitch semibold 1.2))
        (3 . (rainbow variable-pitch semibold 1.1))
        (t . (monochrome variable-pitch 1.1)))
)
``` 

Then, I change on `modus-themes.el` as shown in the following diff:

``` diff
@@ -3228,7 +3228,7 @@ that combines well with the background and foreground."
          (style (or key (modus-themes--key-cdr t modus-themes-headings)))
          (style-listp (listp style))
          (properties style)
-         (var (when (memq 'variable-pitch properties) 'variable-pitch))
+         (var (when (memq 'variable-pitch properties) 'fixed-pitch))
          (varbold (if var
                       (append (list 'bold) (list var))
                     'bold))
```

Thanks for reading this. I admire your work.

**Comment by protesilaos** [comment]
Hello @rolojf and thank you for sharing this!

What we could do in this case is provide a `fixed-pitch` option for `modus-themes-headings`.  This way we get the desired behaviour.

Note that I am in the process of developing version 4 of the themes (the `version-4` branch), which brings lots of changes.  Any change here will be part of that release.

**Comment by rolojf** [comment]
Changed it to fixed pitch as recommended. As follows on my config:

``` elisp
      modus-themes-headings
      '((1 . (overline fixed-pitch bold 1.2))
        (2 . (rainbow fixed-pitch semibold 1.2))
        (3 . (rainbow fixed-pitch semibold 1.1))
        (t . (monochrome fixed-pitch 1.1)))
```

And with this change in modus-themes.el
``` diff
@@ -1296,7 +1296,8 @@ Optional OL is the color of an overline."
          (style (or key (alist-get t modus-themes-headings)))
          (style-listp (listp style))
          (properties style)
-         (var (when (memq 'variable-pitch properties) 'variable-pitch))
+         (var (if (memq 'variable-pitch properties) 'variable-pitch
+                (when (memq 'fixed-pitch properties) 'fixed-pitch)))
          (weight (when style-listp (modus-themes--weight style))))
     (list :inherit
           (cond
```
